File: January/jan18/jan18.java
class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

public class LinkedList {
    public Node reverseList(Node head) {
        Node prev = null; // Initialize previous pointer
        Node curr = head; // Start with the head of the list

        while (curr != null) {
            Node nextNode = curr.next; // Store the next node
            curr.next = prev;          // Reverse the current node's pointer
            prev = curr;               // Move the previous pointer forward
            curr = nextNode;           // Move the current pointer forward
        }

        return prev; // New head of the reversed list
    }
}

File: January/jan18/jan18.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

def reverseList(head):
    prev = None  # Initialize previous pointer
    curr = head  # Start with the head of the list

    while curr is not None:
        next_node = curr.next  # Store the next node
        curr.next = prev       # Reverse the current node's pointer
        prev = curr            # Move the previous pointer forward
        curr = next_node       # Move the current pointer forward

    return prev  # New head of the reversed list
